python/tvm/relay/transform/fake_quantization_to_integer.py

```diff
@@ -158,7 +159,10 @@ def avgpool2d(expr, type_map):
     else:
         arg = relay.op.cast(arg, "int32")
     out = relay.op.nn.avg_pool2d(arg, **expr.attrs)
-    return [out, TensorAffineType(out_t.scale, out_t.zero_point, "int32", out_t.axis)]
+    # Cast back to input dtype to preserve input dtype == output dtype.
+    out = relay.op.clip(out, a_min=np.iinfo(t.dtype).min, a_max=np.iinfo(t.dtype).max)
+    out = relay.op.cast(out, t.dtype)
```

Comment:
Why do you need to preserve `input dtype == output dtype`? As I understood it should match `out_t.dtype`.

I guess it should be conditional injection of cast op to `out_t.dtype` if it is not a `int32`.

Reply:
Make sense. Changed on `out_t.dtype` + if condition.